ops/testing.py

```diff
@@ -107,10 +90,6 @@
     })
     RawConfig = TypedDict("RawConfig", {'options': Dict[str, _ConfigOption]})
 
-# Toggles Container.can_connect simulation globally for all harness instances.
-# For this to work, it must be set *before* Harness instances are created.
-
-SIMULATE_CAN_CONNECT = False
```

Comment:
One thing I was wondering is if we want to leave some sort of property here so that if someone tries to "SIMULATE_CAN_CONNECT=True" then they get an error/warning/something rather than it just being silently ignored.
I know you can do that with class properties, I'm not sure if there is anything you can do for module attributes.
We don't have to, and arguably all we're doing is changing the default to True always, and likely nobody is actively setting it to False. It would be nice to have a way to help people clean up their code. (and I don't think they'll be writing their charm to be compatible with ops1.5 *and* 2.0)

Reply:
Python 3.7 actually added [module-level `__getattr__`](https://docs.python.org/3/reference/datamodel.html#customizing-module-attribute-access) ([PEP 562](https://peps.python.org/pep-0562/)), but it's only for reads if the attribute lookup fails. I don't think it's directly supported for assigning an attribute, though the docs show an example of updating `sys.modules` with a `ModuleType` subclass that adds `__setattr__` as well ... however, it's too clever/magical for my taste, and I don't think it's worth it.

We'll highlight this prominently in the release notes (and update our Discourse docs).